mochajs/mocha
# Type "rs\n" to restart tests, fixes #871

### Description of the Change

Add a keyboard shortcut (`rs\n`) for manually restarting tests when running the CLI in watch (`--watch`) mode.Although the `--watch` feature is somewhat controversial (see #1780), hopefully it's okay to make minor improvements while that debate is being hashed out? 

To test this feature:
1. `mocha -w` to run CLI in watch mode
2. Type [r] [s] [Enter] 
3. Notice that tests restart

### Alternate Designs

This feature mimics the behavior of the popular `nodemon` module, so should be familiar to many developers.  (And, in fact, the code is cribbed pretty much straight from nodemon)

### Why should this be in core?

This is a minor tweak to behavior already in core (`--watch`).  Although the watch feature is somewhat controversial (see #1780), hopefully there's no reason not to make minor improvements as long as it's supported.

### Benefits

Provides behavior that many developers are familiar with in similar tool(s) - specifically `nodemon`.

### Possible Drawbacks

This may make using stdin in the CLI (watch mode) problematic in the future, however that seems like a pretty abstract concern, and mocha does not appear to do anything with stdin [at this time](https://github.com/mochajs/mocha/search?q=stdin&type=Code)

### Applicable issues

None that I'm aware of.

This is an enhancement (semver "minor" release)
